src/prpy/util.py

```diff
@@ -710,29 +716,56 @@ def IsAtTrajectoryStart(robot, trajectory):
     @returns: True if the robot's active DOFs match the given trajectory
               False if one or more active DOFs differ by DOF resolution
     """
-    # Get used indices and starting configuration from trajectory.
     cspec = trajectory.GetConfigurationSpecification()
-    dof_indices, _ = cspec.ExtractUsedIndices(robot)
-    traj_values = cspec.ExtractJointValues(
-        trajectory.GetWaypoint(0), robot, dof_indices)
-
-    # Get current configuration of robot for used indices.
-    with robot.GetEnv():
-        robot_values = robot.GetDOFValues(dof_indices)
-        dof_resolutions = robot.GetDOFResolutions(dof_indices)
-
-    # Check deviation in each DOF, using OpenRAVE's SubtractValue function.
-    dof_infos = zip(dof_indices, traj_values, robot_values, dof_resolutions)
-    for dof_index, traj_value, robot_value, dof_resolution in dof_infos:
-        # Look up the Joint and Axis of the DOF from the robot.
-        joint = robot.GetJointFromDOFIndex(dof_index)
-        axis = dof_index - joint.GetDOFIndex()
-
-        # If any joint deviates too much, return False.
-        delta_value = abs(joint.SubtractValue(traj_value, robot_value, axis))
-        if delta_value > dof_resolution:
+    needs_base = HasAffineDOFs(cspec)
+    needs_joints = HasJointDOFs(cspec)
+    
+    if needs_base and needs_joints:
+        raise ValueError('Trajectories with affine and joint DOFs are not supported')
+
+    if needs_base:
+        doft = openravepy.DOFAffine.X | openravepy.DOFAffine.Y | openravepy.DOFAffine.RotationAxis
+        current_pose = openravepy.RaveGetAffineDOFValuesFromTransform(robot.GetTransform(), doft)
+        start_transform = numpy.eye(4)
+        start_transform = cspec.ExtractTransform(start_transform, trajectory.GetWaypoint(0), robot)
+        traj_start = openravepy.RaveGetAffineDOFValuesFromTransform(start_transform, doft)
+
+        # Compare translation distance
+        trans_delta_value = abs(current_pose[:2] - traj_start[:2])
+        trans_resolution = robot.GetAffineTranslationResolution()[:2]
+        if trans_delta_value[0] > trans_resolution[0] or \
+           trans_delta_value[1] > trans_resolution[1]:
+            return False
+        
+        # Compare rotation distance
+        rot_delta_value = (current_pose[2] - traj_start[2] + numpy.pi) % (2.*numpy.pi) - numpy.pi
```

Comment:
Also, someone needs to make a decision about what this angle wrapping should be, because we are already having issues where some of the code wraps to `[pi, pi]` while other parts wrap to `[-pi,pi)`, while I think `(-pi,pi]` is probably the most intuitive non-redundant interval.

See: https://github.com/personalrobotics/prpy/issues/190